javascript/readme.md

```diff
@@ -1,3 +1,386 @@
+
+```javascript
+var integer = function (
+    value,
+    default_value
+) {
+    value = resolve(value);
+    return (typeof value === "number")
+        ? Math.floor(value)
+        : (typeof value === "string")
+            ? value.charCodeAt(0)
+            : default_value;
+};
+```
+
+If `.` period is the first character on a line, the indentation is increased by 4 spaces.
+
+Avoid excessively long lines. When a statement will not fit nicely on a single line, it may be necessary to break it. It is best to break after a `{` left brace, `[` left bracket, `(` left paren, `,` comma, or before a `.` period, `?` question mark, or `:` colon. If such a break is not feasible, then break after an operator and continue on the next line with 8 spaces added to the current indentation. Those 8 spaces do not change the current indentation.
+
+Clauses (`case`, `catch`, `default`, `else`, `finally`) are not statements and so should not be indented like statements.
+
+Tabs and spaces should not be mixed. We should pick just one in order to avoid the problems that come from having both. Personal preference is an extremely unreliable criteria. Neither the tab nor the space offers a powerful advantage over the other. Fifty years ago, tab had the advantage of consuming less memory, but Moore's Law has eliminated that advantage. Space has one clear advantage over tab: there is still no reliable standard for how many spaces a tab represents, but it is universally accepted that a space occupies a space. So always use spaces. You can edit with tabs if you must, but make sure it is spaces again before you commit. Maybe someday we will finally get a universal standard for tabs, but until that day comes, the better choice is spaces.
+
+## Comments
+
+Be generous with comments. It is useful to leave information that will be read at a later time by people (possibly your future self) who will need to understand what you have done and why. The comments should be well-written and clear, just like the code they are annotating. An occasional nugget of humor might be appreciated. Frustrations and resentments will not.
+
+It is important that comments be kept up-to-date. Erroneous comments can make programs even harder to read and understand.
+
+Make comments meaningful. Focus on what is not immediately visible. Don't waste the reader's time with stuff like
+
+```javascript
+i = 0; // Set i to zero.
+```
+Generally use line comments. Save block comments for formal documentation.
+
+## Variable Declarations
+
+All variables should be declared before used. JavaScript does not require this, but doing so makes the program easier to read and makes it easier to detect undeclared variables that may become implied. Implied global variables should never be used. Use of global variables should be minimized.
```

Comment:
All variables should be declared before used. -> All variables should be declared before use.